runtime/bcutil/cfreader.c

```diff
@@ -655,24 +681,55 @@ readAttributes(J9CfrClassFile * classfile, J9CfrAttribute *** pAttributes, U_32
 				 * to indicate that the attribute is malformed.
 				 */
 				Trc_BCU_MalformedTypeAnnotation(address);
-				if (!ALLOC_ARRAY(annotations->rawAttributeData, length + 1, U_8)) {
+
+				/* calculate the length needed to insert the illegal type_annotation.
+				 * The minimum number of bytes needed is three for
+				 * num_annotations and target_type
+				 */
+				if (1 == length) {
+					/* Treat one byte as part of the type_annotation to reach the correct index. */
+					annotations->rawDataLength = 4;
+				} else if (length < 3) {
+					annotations->rawDataLength = 3;
+				} else {
+					/* Extra slot for illegal target_type */
+					annotations->rawDataLength = length + 1;
```

Comment:
It might be clearer to have a minimum number of byes required and take the `max(min_required_bytes, length + 1)`